core/src/main/java/com/digitalpebble/stormcrawler/spout/FileSpout.java

```diff
@@ -138,9 +132,6 @@ protected void populateBuffer() throws IOException {
                                     StandardCharsets.UTF_8));
         }
 
-        // no more files to read from
-        if (currentBuffer == null) return;
```

Comment:
We set the buffer in line 128 if it is null by a non-null-operation. Therefore the current buffer can never be `null` between line 128 and 145.
Or is `populateBuffer` of a single object instance called by multiple threads? Then we have to do something about threadsafety.

Reply:
there is no worry about thread-safety here. My concern was that it could still be null if 

            _currentBuffer =
                    new BufferedReader(
                            new InputStreamReader(
                                    new FileInputStream(inputPath.toFile()),
                                    StandardCharsets.UTF_8));_
                                    
did not resolve i.e. the input path was incorrect but then we'd get an Exception and probably would not make it to line 142 anyway